test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
@stevekuznetsov 

Just so I am clear, the logic hasn't changed from before right?  You as just asking whether it should change now that we know about the new distinct states?

It's also updating when you aren't waiting in any of these three states right?  Which I would think they are all valid.

@jolamb I don't love having to do the include? here.  I think I would prefer an overarching waiting state to exist.  Or at a min I would abstract the ability to tell that into a method.